internal/cluster/distribution/eks/eksprovider/driver/cluster_creator.go

```diff
@@ -315,6 +320,16 @@ func (c *EksClusterCreator) validate(r *pkgCluster.CreateClusterRequest, logger
 		return errors.WrapIf(err, "failed to create AWS session")
 	}
 
+	// Validate encryption configuration.
+	if modelCluster.EncryptionConfig != nil {
```

Comment:
`len > 0` sounds more appropriate, since this could be a non-nil, 0 length slice, couldn't it?

Reply:
Well those are different cases.

Here we decide whether we want to validate the encryption configuration at all.
Specifying nil encryption configuration is a valid use case when the user opts out of envelope encryption, in such case s we run no validation, this conditional branching checks that case.

The check for the list not being empty is implemented in the validation function's logic as in case of any encryption config being specified we do not want to accept empty configurations.

What do you think, should we approach this `non-existent OK, otherwise validate` situation differently?

Reply:
See my [update](https://github.com/banzaicloud/pipeline/pull/3117#issuecomment-674568253), I convinced myself it is better as I described in the last part of my last comment.